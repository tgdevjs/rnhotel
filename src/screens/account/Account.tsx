import React from "react";
import PropTypes from "prop-types";
import { compose } from "react-apollo";
import { NavigationScreenProp } from "react-navigation";

import {
  withUserMutation,
  withUserQuery
} from "../../apollo/client-state/user";
import { AccountGuest, AccountMenuWithQuery } from ".";
import { UserType } from "../../types";

type Props = {
  navigation: NavigationScreenProp<any, any>;
  user: UserType;
};

export class Account extends React.PureComponent<Props> {
  static navigationOptions = () => ({ headerTitle: "Account" });

  onPress = (screen: string) => {
    this.props.navigation.navigate(screen);
  };

  render() {
    const {
      navigation: { navigate },
      user
    } = this.props;

    if (user.name) {
      return <AccountMenuWithQuery onPress={this.onPress} user={user} />;
    }

    return <AccountGuest onPress={this.onPress} />;
  }
}

export const AccountWithQuery = compose(
  withUserMutation,
  withUserQuery
)(Account);
