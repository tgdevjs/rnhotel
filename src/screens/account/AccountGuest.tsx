import React from "react";
import { Button, StyleSheet, View } from "react-native";
import PropTypes from "prop-types";
import Ionicons from "react-native-vector-icons/MaterialIcons";

import { SignInButton } from "../../components";

type Props = {
  onPress: (screen: string) => void;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  buttonContainer: {
    marginTop: 80
  },
  separator: {
    height: 20
  }
});

export const AccountGuest = ({ onPress }: Props) => {
  return (
    <View style={styles.container}>
      <View>
        <Ionicons name="person" size={80} color="blue" />
        <View style={styles.buttonContainer}>
          <SignInButton onPress={() => onPress("SignIn")} />
          <View style={styles.separator} />
          <Button title="Join" onPress={() => onPress("Join")} />
        </View>
      </View>
    </View>
  );
};
