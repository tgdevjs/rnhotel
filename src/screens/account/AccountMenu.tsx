import React from "react";
import { Button, StyleSheet, Text, View } from "react-native";

import { withUserMutation } from "../../apollo/client-state/user";
import { UserType } from "../../types";
import { fonts } from "../../styles";

type Props = {
  setUser: (user: UserType) => void;
  user: UserType;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  main: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  labelText: fonts.Body,
  valueText: fonts.H3,
  buttonSection: {
    marginBottom: 40,
    justifyContent: "center",
    alignItems: "center"
  }
});

export const AccountMenu = ({ setUser, user: { name } }: Props) => {
  return (
    <View style={styles.container}>
      <View style={styles.main}>
        <Text style={styles.labelText}>User Name:</Text>
        <Text style={styles.valueText}>{name}</Text>
      </View>
      <View style={styles.buttonSection}>
        <Button
          title="SignOut"
          onPress={() => setUser({ variables: { input: { name: null } } })}
        />
      </View>
    </View>
  );
};

export const AccountMenuWithQuery = withUserMutation(AccountMenu);
