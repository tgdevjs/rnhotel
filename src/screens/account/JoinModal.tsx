import React from "react";
import { Alert, Button, StyleSheet, TextInput, View } from "react-native";
import { compose } from "react-apollo";
import { NavigationScreenProp } from "react-navigation";

import { withUserMutation } from "../../apollo/client-state/user";
import { withAddUserMutation } from "../../apollo/client-state/userList";
import { borderRadius, colors, spacing } from "../../styles";

type Props = {
  navigation: NavigationScreenProp<any, any>;
  setUser: () => void;
  addUser: () => void;
};

type State = {
  nameInput: string;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  nameInput: {
    height: 40,
    width: 200,
    borderColor: colors.gray,
    borderWidth: 1,
    borderRadius: borderRadius.small,
    margin: spacing,
    paddingHorizontal: spacing
  }
});

export class JoinModal extends React.PureComponent<Props, State> {
  state = { nameInput: "" };

  onJoin = () => {
    const {
      addUser,
      navigation: { goBack },
      setUser
    } = this.props;
    const { nameInput: name } = this.state;

    if (name === "") {
      Alert.alert("Not valid name", "please input name");
    } else {
      setUser({ variables: { input: { name } } });
      addUser({ variables: { input: { name } } });
      goBack();
    }
  };

  render() {
    const {
      navigation: { goBack }
    } = this.props;
    const { nameInput } = this.state;
    return (
      <View style={styles.container}>
        <TextInput
          onChangeText={text => this.setState({ nameInput: text })}
          onSubmitEditing={this.onJoin}
          placeholder="name"
          style={styles.nameInput}
          value={nameInput}
        />
        <Button title="Join" onPress={this.onJoin} />
        <Button title="Cancel" onPress={() => goBack()} />
      </View>
    );
  }
}

export const JoinModalWithQuery = compose(
  withAddUserMutation,
  withUserMutation
)(JoinModal);
