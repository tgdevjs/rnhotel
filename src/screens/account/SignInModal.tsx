import React from "react";
import {
  Button,
  FlatList,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from "react-native";
import { compose } from "react-apollo";
import { NavigationScreenProp } from "react-navigation";

import { withUserListQuery } from "../../apollo/client-state/userList";
import { withUserMutation } from "../../apollo/client-state/user";

type Props = {
  navigation: NavigationScreenProp<any, any>;
  setUser: () => void;
  userList: [];
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center"
  },
  header: {
    height: 160,
    alignItems: "center",
    justifyContent: "center"
  },
  headerText: {
    fontSize: 24,
    fontWeight: "500",
    marginBottom: 20
  },
  actionText: {
    fontSize: 16
  },
  list: {
    height: "70%"
  },
  itemSeparator: {
    backgroundColor: "black",
    height: 1
  },
  item: {
    justifyContent: "center",
    height: 60
  },
  itemTitleText: {
    marginLeft: 40
  },
  footer: {
    height: 60
  }
});

export const SignInModal = ({
  navigation: { goBack },
  setUser,
  userList
}: Props) => {
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.headerText}>SignIn</Text>
        <Text style={styles.actionText}>Select a user account.</Text>
      </View>
      <FlatList
        style={styles.list}
        data={userList} 
        keyExtractor={({ name }) => name}
        renderItem={({ item: { name } }) => (
          <TouchableOpacity
            style={styles.item}
            onPress={() => {
              setUser({ variables: { input: { name } } });
              goBack();
            }}
          >
            <Text style={styles.itemTitleText}>{name}</Text>
          </TouchableOpacity>
        )}
        ItemSeparatorComponent={() => <View style={styles.itemSeparator} />}
      />
      <Button title="Cancel" onPress={() => goBack()} />
      <View style={styles.footer} />
    </View>
  );
};

export const SignInModalWithQuery = compose(
  withUserListQuery,
  withUserMutation
)(SignInModal);
