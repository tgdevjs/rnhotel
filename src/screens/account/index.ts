export * from './Account';
export * from './AccountGuest';
export * from './AccountMenu';
export * from './JoinModal';
export * from './SignInModal';
