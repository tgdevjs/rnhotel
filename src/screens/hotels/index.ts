export * from './Hotels';
export * from './ConfirmReservation';
