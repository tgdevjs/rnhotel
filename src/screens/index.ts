export * from './account';
export * from './hotels';
export * from './search';
export * from './stays';
