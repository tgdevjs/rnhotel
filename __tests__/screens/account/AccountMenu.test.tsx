import React from "react";
import { AccountMenu } from "../../../src/screens/account";

describe("AccountMenu", () => {
  it("renders default", () => {
    expect(shallow(<AccountMenu user={{ name: "hello" }} />)).toMatchSnapshot();
  });
});
