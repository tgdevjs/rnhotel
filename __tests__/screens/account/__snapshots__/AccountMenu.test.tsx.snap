// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountMenu renders default 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      User Name:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      hello
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 40,
      }
    }
  >
    <Button
      onPress={[Function]}
      title="SignOut"
    />
  </View>
</View>
`;
